### PR TITLE
feat(`npm`): suggest global deps if global flag is specified

### LINKED ISSUE #616: [npm]: correctly suggest packages when using `-g` flag
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [x] I would like to work on this.

### What CLI tool does this relate to?

npm

### Which statement makes the most sense?

This completion spec isn't complete

### Issue Details

When using for example `npm -g rm <package>` we do not provide the correct suggestion.

### PR DESCRIPTION
**Pull request by fedeci**
Closes #616

### CODE REVIEW

File: src/npm.ts [hunk 1]
```diff
@@ -1,3 +1,15 @@
+function uninstallSubcommand(named: string | string[]): Fig.Subcommand {
```
Comment (QuiiBz) [inline src/npm.ts:1]: Any reason why using `named` instead of `name`?

File: src/npm.ts [hunk 2]
```diff
@@ -53,14 +65,13 @@ const workspaceGenerator: Fig.Generator = {
...
-    "until [[ -f package.json ]] || [[ $PWD = '/' ]]; do cd ..; done; cat package.json",
-  postProcess: function (out, context = []) {
-    if (out.trim() === "") {
-      return [];
-    }
-
-    try {
+  trigger: (newToken) => newToken === "-g" || newToken === "--global",
+  custom: async function (tokens, executeShellCommand) {
+    console.error(tokens);
```
Comment (QuiiBz) [inline src/npm.ts:70]: We can remove this `console.error`

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Amazing feature! Would it be possible to port it to `yarn` and `pnpm`?

**Comment by fedeci** [comment]
Do yarn and pnpm allow global commands?

File: src/npm.ts (hunk 2 above)
Comment (fedeci) [inline src/npm.ts:70]:
```suggestion
```

**Comment by QuiiBz** [comment]
> Do yarn and pnpm allow global commands?

They install binaries the same way as `npm` do, that would be helpful for `yarn global remove ...` & `pnpm remove --global ...`

**Comment by fedeci** [comment]
pnpm seems to be already using the npm generator

**Comment by QuiiBz** [comment]
> pnpm seems to be already using the npm generator. Yarn already had the feature.

Nice to hear! But it seems like `yarn` isn't working properly. We might handle this in another PR.

Screenshot 2021-09-30 at 21 17 24

**Comment by fedeci** [comment]
It is working for me🤔 Try to install a package e.g. express globally and see if works

**Comment by QuiiBz** [comment]
Found why it's not working: the generator uses `cat`, which I aliased to [`bat`](https://github.com/sharkdp/bat), and the output isn't the same. The simplest way to fix this is to append a `\` to the command, so the shell will execute the right binary, and not the aliased one. I'm going to open a PR since this issue is happening in multiple specs.

**Review by QuiiBz — APPROVED** [review]
LGTM!

**Pull request merged**